### DISCUSSION TITLE
make the application portable

### DISCUSSION THREAD

**Discussion by HDInfautre** [FreeTubeApp/FreeTube#5435] [Q&A] [answered]
Hello
I realized that I had installed the FreeTube application twice by mistake (flatpak and appcimage) and I now have 2 databases.
My intention is to have a portable application with its database on an external hard drive (to use it on another computer).
For this I put the tree in place like many other appImage but for freetub, there is no file or folder under Freetube.AppImage.home

```
/media/rv/donnees/Com. informat./Appli-donnees-port/Freetube/
/Freetube.AppImage.home/
Freetube.AppImage
```

How to make freetube portable? thank you for the tutorial that I didn't see in the FAQ

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Related issue https://github.com/FreeTubeApp/FreeTube/issues/746, maybe you can find some useful info in there

**Comment by HDInfautre** [comment]
HI
Thanks, I saw it but it concerns Windows.
I think the appimage is not programmed the same.

Afterwards I just did a test by putting all my favorites in the appimage and I have the impression that it is independent, perhaps it is stored in the appimage now?.
I can't test it on another computer at the moment
thank you for confirming

**Comment by Samueru-sama** [comment]
In this case you have to make a `Freetube.AppImage.config` directory and place it next to the AppImage. 

The directory has to havet the same name as the .AppImage file + `.config` at the end.

**Reply by HDInfautre** [reply]
HI
Thank you for your reply.
I've done this with .home for many appImages.

In this case, however, if I move the appImage, I don't lose anything!
So I was wondering where it was stored. so are you sure this is necessary? well with .config?

**Reply by Samueru-sama** [reply]
I just tested adding the `.home` directory to the freetube appimage and it didn't work, it was still using my home config directory and the fakehome only had a `.pki` directory.

Making a `Freetube.AppImage.config` instead fixed that issue, it's weird but it works. 

Since freetube is an electron application, you only really need to store the files from its config directory, since this is a bug that chromium has that it stores everything in its config dir lol

**Comment by HDInfautre** [comment, marked as answer]
HI
In the end I succeeded with this superb application, which is full of promise https://github.com/mijorus/gearlever because I didn't manage with the other solutions.

**Reply by Samueru-sama** [reply]
[Did you try this one? ](https://github.com/ivan-hc/AM)

edit: nvm, I didn't know you were the same person from keepassxc lol

**Reply by HDInfautre** [reply]
I didn't get there, it's more complex than the one I recommend.

**Reply by Samueru-sama** [reply]
> I didn't get there, it's more complex than the one I recommend.

Really? because with the other one if you want to make a portable home or config it is as simple as running `am --home freetube` and/or `am --config freetube` and that is it.

It also has a internal database of appimages that includes freetube, you would install it by running `am -i freetube` and it also automatically updates the appimage, it also has a feature that lets you make appimages change to the static runtime and that way you don't have to be installing libfuse2 in recent versions of ubuntu that don't come with it. 

The only problem might be that `am` only installs in `/opt`, there is a another mode called `appman` that lets you install appimages in in your home, to do so in a external hdd you would need to make a symlink in your home to that hard-drive and then tell appman to install the appimages in that symlink.
